File: urbanairship-sdk/src/test/java/com/urbanairship/json/JsonListTest.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.json;

import com.urbanairship.BaseTestCase;

import org.json.JSONException;
import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNotNull;
import static junit.framework.Assert.assertTrue;


public class JsonListTest extends BaseTestCase {

    private JsonList jsonList;

    @Before
    public void setUp() throws JsonException {
        jsonList = JsonValue.wrap(new Object[] { "first-value", "second-value", null }).getList();
        assertNotNull(jsonList);
    }

    /**
     * Test creating a new JsonList with a null list.
     */
    @Test
    public void testCreateNull() throws JsonException, JSONException {
        JsonList emptyList = new JsonList(null);
        assertEquals(0, emptyList.size());
        assertTrue(emptyList.isEmpty());
    }

    /**
     * Test toString produces a JSON encoded String.
     */
    @Test
    public void testToString() {
        String expected = "[\"first-value\",\"second-value\"]";
        assertEquals(expected, jsonList.toString());
    }

    /**
     * Test toString on an empty list produces a JSON encoded String.
     */
    @Test
    public void testEmptyMapToString() {
        assertEquals("[]", new JsonList(null).toString());
    }
}

File: urbanairship-core/src/test/java/com/urbanairship/json/JsonListTest.kt
/* Copyright Airship and Contributors */
package com.urbanairship.json

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.urbanairship.json.JsonValue.Companion.wrap
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
public class JsonListTest {

    private var jsonList: JsonList? = null

    @Before
    @Throws(JsonException::class)
    public fun setUp() {
        jsonList = wrap(arrayOf("first-value", "second-value", null)).list
        Assert.assertNotNull(jsonList)
    }

    /**
     * Test creating a new JsonList with a null list.
     */
    @Test
    public fun testCreateNull() {
        val emptyList = JsonList(null)
        Assert.assertEquals(0, emptyList.size())
        Assert.assertTrue(emptyList.isEmpty)
    }

    /**
     * Test toString produces a JSON encoded String.
     */
    @Test
    public fun testToString() {
        val expected = "[\"first-value\",\"second-value\"]"
        Assert.assertEquals(expected, jsonList.toString())
    }

    /**
     * Test toString on an empty list produces a JSON encoded String.
     */
    @Test
    public fun testEmptyMapToString() {
        Assert.assertEquals("[]", JsonList(null).toString())
    }
}
